.github/workflows/.test.yml

```diff
@@ -151,48 +153,28 @@ jobs:
             export TESTFLAGS="${TESTFLAGS} --run=//worker=${{ matrix.worker }}$"
           fi
           ./hack/test ${{ matrix.kind }}
-          mv ./coverage/coverage.txt ./coverage/coverage-${{ github.job }}-$(echo "${{ matrix.pkg }}-${{ matrix.skip-integration-tests }}-${{ matrix.kind }}-${{ matrix.worker }}-${{ matrix.tags }}" | tr -dc '[:alnum:]-\n\r' | tr '[:upper:]' '[:lower:]').txt
         env:
           TEST_COVERAGE: 1
           TESTPKGS: ${{ matrix.pkg }}
           SKIP_INTEGRATION_TESTS: ${{ matrix.skip-integration-tests }}
           CACHE_FROM: type=gha,scope=${{ inputs.cache_scope }}
       -
-        name: Upload coverage file
-        continue-on-error: ${{ matrix.tags == 'nydus' }}
-        uses: actions/upload-artifact@v3
-        with:
-          name: coverage
-          path: ./coverage
-
-  upload-coverage:
-    runs-on: ubuntu-20.04
-    needs:
-      - run
-    steps:
-      -
-        name: Checkout
-        uses: actions/checkout@v3
-      -
-        name: Download coverage files
-        uses: actions/download-artifact@v3
+        name: Send to Codecov
+        if: always()
+        uses: codecov/codecov-action@v3
         with:
-          name: coverage
-          path: ./coverage
+          directory: ./bin/testreports
+          flags: ${{ matrix.codecov_flags }}
       -
-        name: List coverage files
-        id: files
-        uses: actions/github-script@v6
+        name: Generate annotations
+        if: always()
+        uses: crazy-max/.github/.github/actions/gotest-annotations@main
```

Comment:
Using a remote composite action to reduce dedup. Will use a commit sha if we are ok with this implementation.

See https://github.com/crazy-max/.github#gotest-annotations